test/DeltaTriplesTest.cpp

```diff
@@ -0,0 +1,327 @@
+              .toValueId(vocab, localVocab),
+          std::move(triple.object_).toValueId(vocab, localVocab)};
+      return IdTriple<0>(ids);
+    };
+    return ad_utility::transform(
+        makeTurtleTriples(std::move(turtles)),
+        [&toID](TurtleTriple triple) { return toID(std::move(triple)); });
+  }
+};
+
+// Test clear after inserting or deleting a few triples.
+TEST_F(DeltaTriplesTest, clear) {
+  auto cancellationHandle =
+      std::make_shared<ad_utility::CancellationHandle<>>();
+
+  DeltaTriples deltaTriples(testQec->getIndex());
+  auto& vocab = testQec->getIndex().getVocab();
+  auto& localVocab = deltaTriples.localVocab();
+
+  ASSERT_EQ(deltaTriples.numInserted(), 0);
+  ASSERT_EQ(deltaTriples.numDeleted(), 0);
+  EXPECT_THAT(deltaTriples, m::NumTriplesInAllPermutations(0));
+
+  // Insert then clear.
+  deltaTriples.insertTriples(
+      cancellationHandle, makeIdTriples(vocab, localVocab, {"<a> <UPP> <A>"}));
+
+  ASSERT_EQ(deltaTriples.numInserted(), 1);
+  ASSERT_EQ(deltaTriples.numDeleted(), 0);
+  EXPECT_THAT(deltaTriples, m::NumTriplesInAllPermutations(1));
+
+  deltaTriples.clear();
+
+  ASSERT_EQ(deltaTriples.numInserted(), 0);
+  ASSERT_EQ(deltaTriples.numDeleted(), 0);
+  EXPECT_THAT(deltaTriples, m::NumTriplesInAllPermutations(0));
+
+  // Delete, insert and then clear.
+  deltaTriples.deleteTriples(
+      cancellationHandle, makeIdTriples(vocab, localVocab, {"<A> <low> <a>"}));
+  ASSERT_EQ(deltaTriples.numInserted(), 0);
+  ASSERT_EQ(deltaTriples.numDeleted(), 1);
+  EXPECT_THAT(deltaTriples, m::NumTriplesInAllPermutations(1));
+
+  deltaTriples.insertTriples(
+      cancellationHandle, makeIdTriples(vocab, localVocab, {"<a> <UPP> <A>"}));
+
+  ASSERT_EQ(deltaTriples.numInserted(), 1);
+  ASSERT_EQ(deltaTriples.numDeleted(), 1);
+  EXPECT_THAT(deltaTriples, m::NumTriplesInAllPermutations(2));
+
+  deltaTriples.clear();
+
+  ASSERT_EQ(deltaTriples.numInserted(), 0);
+  ASSERT_EQ(deltaTriples.numInserted(), 0);
+  EXPECT_THAT(deltaTriples, m::NumTriplesInAllPermutations(0));
+}
+
+TEST_F(DeltaTriplesTest, insertTriplesAndDeleteTriples) {
+  DeltaTriples deltaTriples(testQec->getIndex());
+  auto& vocab = testQec->getIndex().getVocab();
+  auto& localVocab = deltaTriples.localVocab();
+
+  auto cancellationHandle =
+      std::make_shared<ad_utility::CancellationHandle<>>();
+
+  auto mapKeys = [](auto& map) {
+    return ad_utility::transform(map,
+                                 [](const auto& item) { return item.first; });
+  };
+
+  ASSERT_EQ(deltaTriples.numInserted(), 0);
+  ASSERT_EQ(deltaTriples.numDeleted(), 0);
+  EXPECT_THAT(deltaTriples, m::NumTriplesInAllPermutations(0));
+  EXPECT_THAT(
+      mapKeys(deltaTriples.triplesInserted_),
+      testing::UnorderedElementsAreArray(makeIdTriples(vocab, localVocab, {})));
+  EXPECT_THAT(
+      mapKeys(deltaTriples.triplesDeleted_),
+      testing::UnorderedElementsAreArray(makeIdTriples(vocab, localVocab, {})));
```

Comment:
This whole block can become a stateful matcher that captures your `vocab` and `localVocab`.